util/src/network/handshake.rs

```diff
@@ -128,31 +128,26 @@ impl Handshake {
 	/// Readable IO handler. Drives the state change.
 	pub fn readable<Message>(&mut self, io: &IoContext<Message>, host: &HostInfo) -> Result<(), UtilError> where Message: Send + Clone {
 		if !self.expired() {
-			match self.state {
-				HandshakeState::New => {}
-				HandshakeState::ReadingAuth => {
-					if let Some(data) = try!(self.connection.readable()) {
+			while let Some(data) = try!(self.connection.readable()) {
+				match self.state {
+					HandshakeState::New => {},
+					HandshakeState::StartSession => {},
+					HandshakeState::ReadingAuth => {
 						try!(self.read_auth(io, host.secret(), &data));
-					};
-				},
-				HandshakeState::ReadingAuthEip8 => {
-					if let Some(data) = try!(self.connection.readable()) {
+					},
+					HandshakeState::ReadingAuthEip8 => {
 						try!(self.read_auth_eip8(io, host.secret(), &data));
-					};
-				},
-				HandshakeState::ReadingAck => {
-					if let Some(data) = try!(self.connection.readable()) {
+					},
+					HandshakeState::ReadingAck => {
 						try!(self.read_ack(host.secret(), &data));
-					};
-				},
-				HandshakeState::ReadingAckEip8 => {
-					if let Some(data) = try!(self.connection.readable()) {
+					},
+					HandshakeState::ReadingAckEip8 => {
 						try!(self.read_ack_eip8(host.secret(), &data));
-					};
-				},
-				HandshakeState::StartSession => {
+					},
+				}
+				if self.state == HandshakeState::StartSession {
```

Comment:
yes but why not place `io.clear_timer(self.connection.token).ok();` in the braces of `HandshakeState::StartSession => {},`?